### PR TITLE
Add collapsibles for answers and questions

### PR DESCRIPTION
**Pull request by pixeldesu**
Desktop:
image

Mobile:
image

On detail pages the content is always fully shown.

Test with your themes and report any issues. Also check with long new questions if the gradient works for you.

### CODE REVIEW

File: app/assets/stylesheets/components/_collapse.scss [hunk 1]
```diff
@@ -0,0 +1,94 @@
+@use "sass:map";
+
+.collapsed {
+  display: -webkit-box;
+  -webkit-box-orient: vertical;
+  overflow: hidden;
+
+  &.answerbox__answer-text {
+    -webkit-line-clamp: 12;
```
Comment (nilsding) [inline app/assets/stylesheets/components/_collapse.scss:9]: I really don't like how a `-webkit-` prefix became a "standard" ...

Comment (pixeldesu) [reply app/assets/stylesheets/components/_collapse.scss:9]: Nothing I can do about every browser implementing this and not implementing `line-clamp`.

File: app/helpers/theme_helper.rb [hunk 2]
```diff
@@ -36,20 +36,29 @@ def render_theme
     theme.attributes.each do |k, v|
       next unless ATTRIBUTE_MAP.key?(k)
 
-      if k.include?("text") || k.include?("placeholder")
-        hex = get_hex_color_from_theme_value(v)
-        body += "\t--#{ATTRIBUTE_MAP[k]}: #{get_decimal_triplet_from_hex(hex)};\n"
+      if ATTRIBUTE_MAP[k].is_a?(Array)
```
Comment (nilsding) [inline app/helpers/theme_helper.rb]: Sick Ruby Tricks™ #­54: use [`Kernel#Array`](https://ruby-doc.org/core-3.1.0/Kernel.html#method-i-Array) and you can simply use `.each` regardless whether the thing is an array or not:

```ruby
Array(ATTRIBUTE_MAP[k]).each do |var|
  body += "\t--#{var}: #{get_color_for_key(var, v)};\n"
end
```

**Review by nilsding — CHANGES_REQUESTED** [review]

**Review by nilsding — APPROVED** [review]

**Comment by raccube** [comment]
image
I'm happy with it but the button is offset due to the text not filling the width. Would have to make the text wrap around the dropdown caret.

**Review by raccube — APPROVED** [review]

**Pull request merged**